unit_testing_guidelines.md

```diff
@@ -0,0 +1,179 @@
+# Unit and Integration Testing guidelines
+
+This document briefly covers testing philosophy with regard to integration and unit tests,
+especially for the Python language and in the context of developing a KBase core server like
+the Execution Engine.
+
+## Unit versus Integration tests
+
+Unit tests cover one module, class, or function, called a code unit from here on out. For example,
+a unit test file might cover the contents of `my_module.py` or more granularly `my_module.MyClass`.
+Code outside the code unit should be excluded from the tests. The exception is "value classes"
+which are classes which primarily hold data and whose behavior is based on that data. Other
+classes required by the unit under test should be mocked out.
+
+This makes unit tests fast and easy to understand, as only the isolated code unit needs to be
+comprehended in order to comprehend test failures.
```

Comment:
swap out one of those 'comprehend's for a synonym - e.g. 'understand'

Reply:
fixed